2025/lab/Keys/Week7_Missingness_Robustness.Rmd

```diff
@@ -0,0 +1,551 @@
+
+```{r some-mods}
+
+mod1 <- lm(logincome ~ prestige + self_employed + race + female,
+           data = gss, weights = wtssnrps, na.action = na.exclude) # The default is to exclude NAs. I wrote it out to show that cases were left out
+
+mod2 <- lm(logincome ~ prestige + self_employed + race + female,
+           data = gss_comp, weights = wtssnrps) # NOTE The use of gss_comp here 
+
+mod3 <- lm(logincome ~ prestige + self_employed + race + female,
+           data = gss_avail, weights = wtssnrps) # NOTE The use of gss_avail here 
+
+ourmods <- list("w/ NAs" = mod1, "Complete-case" = mod2, "Available-case" = mod3)
+
+
+modelsummary(ourmods, 
+             estimate = "{estimate} ({std.error}) {stars}",
+             statistic = NULL,
+             gof_omit = "F|RMSE|Log|IC",
+             output = "huxtable")
+
+```
+
+# Model Diagnostics and Developing Robust Models
+
+## 1. Dealing with Missing Data
+
+We will begin by examining how model coefficients change when we exclude cases with missing data. The `modelsummary::modelplot()` function automates the transformation of our models into `ggplot2` objects, allowing us to visualize these changes easily.
+
+```{r  mod-comp}
+
+modelplot(list("Complete-case" = mod2, "Available-case" = mod3), 
+          coef_omit = 'Interc') +
+  geom_vline(aes(xintercept = 0), 
+             linetype = "dotted", size = 1, alpha = .5)
+
+```
+
+Missing data can bias our results, so it's crucial to understand its patterns and address it accordingly.
+
+### Patterns of Missingness
+To visualize which variables have missing data, we can create an "upset plot" using the naniar package. This plot will show the combination of variables with missing observations.
+
+```{r miss-upset}
+library(naniar)
+
+gg_miss_upset(gss)
+
+```
+
+Notably, `income` and `logincome` are equivalent, so any missing data in one is also missing in the other. The variable `workhrs` also has a significant number of missing values, which, according to the [GSS Data Explorer](https://gssdataexplorer.norc.org/variables/4/vshow), correspond to individuals not working full-time or part-time.
+
+(This website provides other great ways to visualize missingess: <https://cran.r-project.org/web/packages/naniar/vignettes/naniar-visualisation.html>)
+
+What do we do with the missing data? The approach should be driven by our research question and the variables it deems important. As Ricardo has said early on in the semester, we should use a subset of data that have no missing values across all variables of interest and the variables in our anticipated final model -- creating subsets accordingly (e.g., gss_avail). When missingness is random, imputation is a viable option, and the `mice` package provides an excellent set of tools for this purpose. However, if missingness is not random, it will introduce bias in our model and you should seek advice on more advanced techniques to appropriately handle the missing data. 
```

Comment:
@brenthoagland Check this statement and update based on what I describe in lecture (look at slides from last time, I haven't changed much here). Fix mention about Ricardo here.